### ISSUE TITLE
[LINT_BUG]: r6_usage_linter doesn't detect inherited fields

### ISSUE THREAD

**Issue by TymekDev** [Appsilon/box.linters#157]
### box.linters version

0.10.5

### Sample source code to lint

```r
box::use(
  R6[R6Class],
)

foo <- R6Class(
  public = list(
    xyz = 123
  )
)

bar <- R6Class(
  inherit = foo,
  public = list(
    get_xyz = function() {
      self$xyz
    }
  )
)

bar$new()$get_xyz()
```

### Lint command used

```r
lintr::lint(
  "box::use(
  R6[R6Class],
)

foo <- R6Class(
  public = list(
    xyz = 123
  )
)

bar <- R6Class(
  inherit = foo,
  public = list(
    get_xyz = function() {
      self$xyz
    }
  )
)

bar$new()$get_xyz()",
  linters = lintr::linters_with_defaults(defaults = box.linters::rhino_default_linters)
)
```

### Lint result

```
<text>:15:7: warning: [r6_usage_linter] Internal object call not found.
      self$xyz
      ^~~~~~~~
```

### Expected result

No lints raised as the field is inherited from the base class.

**Comment by radbasa** [comment]
Taking a look at this now.

**Comment by radbasa** [comment]
Two ways to inherit a class:
1. inherit a class from the same source file - need to keep track of the public methods of the super class to make them available in the subclass.
2. inherit a class attached using {box} - getting the public methods of an R6 class attached by box should already be handled by existing helper functions

Both of these require file-level linting. `r6_usage_linter()` currently performs expression-level linting. A refactor is necessary.

XPATH to get an R6 class' super class:
```XPath
 //expr[
    SYMBOL and
    preceding-sibling::SYMBOL_SUB[text() = 'inherit']
  ]
```

**Comment by radbasa** [comment]
Handling inheritance will also require some recursion.
